R/MsBackendTimsTof-functions.R

```diff
@@ -138,16 +144,50 @@ MsBackendTimsTof <- function() {
         if (!length(sd <- setdiff(columns, c("frame", "scan"))))
             stop("At least one column value different from 'frame' and",
                  " 'scan' required")
-        tmp <- query(tms, unique(x@indices[I, "frame"]), c("frame", "scan", sd))
-        f <- factor(paste(tmp$frame, tmp$scan),
-                    levels = paste(x@indices[I, "frame"], x@indices[I, "scan"]))
-        if (length(sd) == 1)
-            res[I] <- unname(split(tmp[, sd], f))
-        else res[I] <- unname(split.data.frame(as.matrix(tmp[, sd]), f))
+        ## Ensure mz and intensity are always first, if provided
+        sd <- c(intersect(c("mz", "intensity"), sd),
```

Comment:
This is to ensure that `"mz"` and `"intensity"` are always (if requested) reported as the first two columns.

Reply:
It's for compatibility reasons: there is the unwritten rule that the first column should contain m/z and the second intensity values. While in `Spectra` we access the columns with their actual name, it could be that other developers rely on that. So, this is just to ensure that.

Reply:
Mentioning it in the small documentation above the function wouldn't be wrong I guess but for me it's ok also like it is, it was just a curiosity :) . Should also `spectraData` preserve this order then?

Reply:
no, for `spectraData` it does not really matter. That should be more like a `data.frame` and then the columns should be returned in the order they are requested. I will add a short documentation to that.